### ISSUE TITLE
Ambiguous binding should throw exception instead of return missing in permissive Mode

### ISSUE THREAD

**Issue by yliuuuu** [partiql/partiql-lang-kotlin#705]
In issue #194, we discussed how dynamic resolution should failed instead of returning the first value encountered  in case of ambiguous binding. 

This issue is to extend the discussion.

Consider the following query:
``SELECT a FROM `[{a:2},{b:2}]` as f, `[{a:3},{b:3}]` as g``

According to the spec: the above query should take in the following binding tuple: 
```
<<
< f: {'a':2}, g:{'a':3}>
< f: {'a':2}, g:{'b':3}> 
< f: {'b':2}, g:{'a':3}>
< f: {'b':2}, g:{'b':3}>
>>
```
Current behavior in Permissive mode returns:
```
<<
{'a': 2},       // ambiguous binding return the first one 
{'a': 2},       // a from f
{'a': 3},       // a from g
{}                // no value found, hence missing
>>
``` 

The same query with `StaticTypeVisitorTransform` enabled (assuming no binding named `a` in the global environment) will throw an error: 
```
org.partiql.lang.ast.passes.SemanticException: No such variable named 'a'
	Semantic Error: at line 1, column 8: No such variable named 'a'.
```

Ideally, the "disambiguation" analysis should run regardless of type analysis and presence of global variable (perhaps an individual pass as a transformer), and the error message should be 
```
org.partiql.lang.eval.SemanticException: Multiple matches were found for the specified identifier
         Semantic Error: at line 1, column 8: Binding name was 'a'
```

Bottom Line here is that it should be semantic error, i.e., we do not want the query to execute and running the risk of producing an error after having executed a large number of "rows" already.

**Comment by lziq** [comment]
When mentioning "return missing", were you referring to the behavior of type inferencer or EvaluatingCompiler? If it is EvaluatingCompiler, is the current behavior returning the first value encountered? 

I believe that, in a normal sql language, `SELECT a FROM t1, t2` throws an error, if both `t1` and `t2` contains a column with name of `a`. In PartiQL, my opinion is that if there are 2 columns with the same name in the FROM clause after cross-join, we can't really determine which one the user is referring to. In this case, we should throw an error and ask users to qualify the variable.

**Comment by yliuuuu** [comment]
Fundamentally this issue is the same as issue #347. To keep an explicit note, we should throw a compile time error suggesting ambiguous binding when we could not resolve the variable name `a`, i.e.,
1. we do not have a schema and more then one binding names exists. 
2. we have a schema but `a` could be found in more than one binding names. 

Bottom Line is that we should throw the issue in compile time, perhaps via an error with `ErrorCategory.SEMANTIC`, instead of in evaluating time which we currently do.
